### ISSUE TITLE
Google app engine deploy action stopped working overnight

### ISSUE THREAD

**Issue by mishacajic** [GoogleCloudPlatform/github-actions#191]
### TL;DR

Without any changes on our end, this action stopped working overnight. It fails to deploy to app engine due to a service account error and no change we've tried have fixed it.

**Expected behavior**

App should deploy to app engine normally as it did before.

**Observed behavior**

Action fails at the deploy step with the following error:
```
ERROR: (gcloud.app.deploy) PERMISSION_DENIED: You do not have permission to act as '[project id redacted]@appspot.gserviceaccount.com'
27
- '@type': type.googleapis.com/google.rpc.ResourceInfo
28
  description: You do not have permission to act as this service account.
29
  resourceName: [redacted]@appspot.gserviceaccount.com
30
  resourceType: serviceAccount
```

It seems like now the action attempts to use the default gcloud service account credentials rather than the service account I set up for it, even though I have `export_default_credentials: true`. It doesn't recognise the newly exported credentials.

### Reproduction

**Action YAML**

```yaml
name: Backend staging CI

on:
  push:
    branches: [staging]

jobs:
  deploy:
    runs-on: ubuntu-latest

    env:
      CI: ""

    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
      # Set up authentication to google cloud via the service account
      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
        with:
          service_account_key: ${{ secrets.GCP_SA_KEY }}
          export_default_credentials: true

      - name: Install node dependencies
        run: npm ci
      - name: Deploy to google cloud
        run: |
          gcloud config set project [project id redacted]
          gcloud app deploy api-staging.yaml --quiet

```

Any help would be appreciated. Thanks!

**Comment by elyobo** [comment]
Not many hits for this, but [this StackOverflow issue](https://stackoverflow.com/questions/64236468/cloud-build-fails-to-deploy-to-google-app-engine-you-do-not-have-permission-to) suggests adding the `Service Account User` role. Also trying this (although in my case I was running deploys from CircleCI).

**Comment by ValentinMoullet** [comment]
Hi! Same issue on our side, the automatic deployment started to fail this morning (we use CI/CD on GitLab though). Adding the `Service Account User` role to the service account used for deployment solved the issue, as long as you also have the `App Engine Service Admin` role on it.

If it can help, here is the list of roles we have on the service account used for automatic deployment:
- `App Engine Deployer`
- `App Engine Service Admin`
- `Cloud Build Service Account`
- `Service Account User`
- `Service Usage Viewer`

**Comment by elyobo** [comment]
Getting the right set of roles feels like a black art; this is what I've documented in the past for starting from scratch (added the suddenly-we-need-this-apparently `Service Account User` this afternoon).

Interestingly we didn't need `Service Usage Viewer`, but did need a couple of `Storage Object` roles (I assumed to upload everything into GCS, but :man_shrugging:).

* create a service account, generally naming it "<project name>-ci"
* add the following roles
  * App Engine Deployer
  * App Engine Service Admin
  * Cloud Build Service Account
  * Service Account User
  * Storage Object Creator
  * Storage Object Viewer
* enable the following APIs on your project
  * App Engine Admin API
  * Cloud Build API

**Comment by mishacajic** [comment]
Adding the Service Account User role worked. Really appreciate the quick responses, thanks very much!

**Comment by gameolive-studio** [comment]
Adding ```Service Account User``` role to our CI user resolves the issue for us.

**Comment by EltonCarreiro** [comment]
Just to confirm, same here, added `Service Account User` and it worked, thanks!

**Comment by ChrisRus** [comment]
Not super happy to be spelunking this at the moment, GC. Would have been better to add the new permission and then make sure all user were aware that they should maybe do something explicitly about changing permissions vs just breaking active deployments?

**Comment by muzuiget** [comment]
This breaking change is ridiculous. A "Service Account" requires a "Service Account User" permission. And the error message is misleading.

**Comment by kkaung** [comment]
> Hi! Same issue on our side, the automatic deployment started to fail this morning (we use CI/CD on GitLab though). Adding the `Service Account User` role to the service account used for deployment solved the issue, as long as you also have the `App Engine Service Admin` role on it.
> 
> If it can help, here is the list of roles we have on the service account used for automatic deployment:
> 
> * `App Engine Deployer`
> * `App Engine Service Admin`
> * `Cloud Build Service Account`
> * `Service Account User`
> * `Service Usage Viewer` (probably useless in most cases)

Thank you!  This solved my issue.

**Comment by jonassunandar** [comment]
> Adding `Service Account User` role to our CI user resolves the issue for us.

July 1, 2021. Adding ``Service Account User`` to the role is still the answer. Thank you!

**Comment by jorge-leandro** [comment]
> > Adding `Service Account User` role to our CI user resolves the issue for us.
> 
> July 1, 2021. Adding `Service Account User` to the role is still the answer. Thank you!

May 2, 2022. Adding Service Account User to the role is still the answer. Thank you!

**Comment by WhiskySoda** [comment]
This is ridiculous. Been stuck on this for days now.
